talentsearch/resources/js/components/skills/SkillChecklist.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react";
+import groupBy from "lodash/groupBy";
+import sum from "lodash/sum";
+import { useIntl } from "react-intl";
+import { getLocale } from "@common/helpers/localize";
+import { SkillFamily } from "@common/api/generated";
+import {
+  getSkillCategoryText,
+  getSkillCategoryImage,
+} from "@common/constants/localizedConstants";
+
+interface FamilyProps {
+  family: SkillFamily;
+  checked: boolean;
+  callback: (family: SkillFamily, checked: boolean) => void;
+}
+
+const Family: React.FunctionComponent<FamilyProps> = ({
+  family,
+  checked,
+  callback,
+}) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  const uncheckedStyle = { "data-h2-font-weight": "b(400)" };
+  const checkedStyle = { "data-h2-font-weight": "b(700)" };
+
+  return (
+    <div
+      {...(checked ? checkedStyle : uncheckedStyle)}
+      data-h2-padding="b(all, xxs)"
+      key={family.key}
+    >
+      <input
+        type="checkbox"
+        id={family.id}
+        onChange={(e) => {
+          callback(family, e.target.checked);
+        }}
+      />
+      &nbsp;
+      <label htmlFor={family.id}>
+        {family.name?.[locale]} ({family.skills ? family.skills.length : 0})
+      </label>
+    </div>
+  );
+};
+
+interface CategoryProps {
+  category: string;
```

Comment:
Does it cause issues if you set this type to be the SkillCategory enum instead of any string? Because really, what we expect is one of those specific enum strings.